packages/design-system/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,21 @@
+import { clsx } from 'clsx';
+import { createElement, forwardRef } from 'react';
+import { textRecipe, TextRecipe } from './Text.css';
+
+export type TextProps = {
+  as?: keyof JSX.IntrinsicElements;
+  className?: string;
+  children: React.ReactNode;
+  dataTestId?: string;
+} & TextRecipe;
+
+export const Text = forwardRef<HTMLElement, React.PropsWithChildren<TextProps>>(
+  ({ variant, weight, children, color, dataTestId, className, as = 'p' }, forwardedRef) => {
+    return createElement(as, {
+      children,
+      className: clsx(textRecipe({ variant, weight, color }), className),
+      ref: forwardedRef,
+      'data-testid': dataTestId,
+    });
+  }
+);
```

Comment:
Can we replace the explicit `createElement` call with JSX?

```suggestion
export const Text = forwardRef<HTMLElement, React.PropsWithChildren<TextProps>>(
  ({ variant, weight, children, color, dataTestId, className, as: Tag = 'p' }, forwardedRef) => {
   return (
     <Tag
       className={...}
       ref={...}
       ...
      >
        {children}
      </Tag> 
   )
  }
);
```